### ISSUE TITLE
expirations are not RFC 3339 strings

### ISSUE THREAD

**Issue by tcmitchell** [GENI-NSF/geni-ch#243]
When returning data in lookup PROJECT or SLICE, the expiration date looks like "2014-03-29 13:49:00", which is not an RFC 3339 compliant date format. It also has an implicit timezone, which is undesirable.

Fix to use RFC 3339 formatting and an explicit timezone e.g. "2014-03-29T13:49:00+00:00".

 _Imported from trac ticket [#243](http://trac.gpolab.bbn.com/chapi/ticket/243),  created by tmitchel on 02-19-2014 at 14:19, last modified: 02-21-2014 at 11:35_

**Comment by tcmitchell** [comment]
Watch for other methods that return this data, those probably need to change as well.

_Trac comment by tmitchel (github user: tcmitchell) on 02-19-2014 at 14:19_

**Comment by MarshallBrinn** [comment]
I made what I think are the complete set of fixes for TKT243 ("expirations are not RFC 3339 strings"). 

In general, CHAPI was returning the 'str' of any datetime returned from the database and not explicitly formatting it. Also, it was not checking that all times passed in to chapi were valid times (parseable by dateutil.parser.parse).

I made changes on this branch and have tested it.

tkt243_rfc3339_datetimes

Awaiting review from Aaron or Tom.

_Trac comment by mbrinn (github user: MarshallBrinn) on 02-20-2014 at 16:57_
